data-analytics/iot-nirvana/pipeline/src/test/java/com/google/cloud/demo/iot/nirvana/pipeline/JsonToEntityTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.demo.iot.nirvana.pipeline;
+
+import static com.google.datastore.v1.client.DatastoreHelper.makeKey;
+import static com.google.datastore.v1.client.DatastoreHelper.makeValue;
+
+import com.google.cloud.demo.iot.nirvana.common.CityTemperature;
+import com.google.cloud.demo.iot.nirvana.common.Message;
+import com.google.cloud.demo.iot.nirvana.common.TemperatureUtils;
+import com.google.datastore.v1.Entity;
+import com.google.datastore.v1.Key;
+import com.google.gson.Gson;
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.beam.sdk.testing.NeedsRunner;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link JsonToEntity}. */
+@RunWith(JUnit4.class)
+public final class JsonToEntityTest {
+
+  @Rule public final transient TestPipeline pipeline = TestPipeline.create();
+
+  private Message messageTirana;
+  private Message messageOranjestad;
+  private CityTemperature cityTirana;
+  private CityTemperature cityOranjestad;
+  private Entity entityTirana;
+  private Entity entityOranjestad;
+
+  /** Create test data. */
+  @Before
+  public void createTestData() throws NoSuchAlgorithmException, UnsupportedEncodingException {
```

Comment:
A lot of these can be static final.

Reply:
I went over all the test data attributes and the only way I see making them static final is to create a static method of initialization for each one individually, which seems overengineering for test classes.